datastore/badger/v4/datastore.go

```diff
@@ -142,8 +142,11 @@ func NewDatastore(path string, options *Options) (*Datastore, error) {
 		gcSleep = gcInterval
 	}
 
-	opt.Dir = path
-	opt.ValueDir = path
+	if !opt.InMemory {
+	    opt.Dir = path
```

Comment:
todo: The linter is complaining, looks like you have tried to sneak spaces, not tabs into our codebase - sorry about the bother, but if you fix this I'll merge your PR immediately after :) :tada: